Removing qt4 capability from faust2{faustvst,jaqt,lv2}

dvzrv opened the pull request:
This removes qt4 functionality from faust2{faustvst,jaqt,lv2} as discussed in #300.
Additionally this fixes many (but not all!) general problems (e.g. unquoted variables, etc.), found using [shellcheck](https://www.shellcheck.net/).

There's still plenty of tab poisoning going on in these files, too, but that can be done in a separate pull request.

It would be awesome, if these scripts could be integration tested during each pull request.

dvzrv committed:
tools/faust2appls/faust2faustvst: Setting shebang properly. Removing qt4 functionality and searching for qmake-qt5/qmake using the shell builtin 'command'.

dvzrv committed:
tools/faust2appls/faust2faustvst: Fixing most missing-quotes related issues in this script.

dvzrv committed:
tools/faust2appls/faust2jaqt: Removing qt4 capability from the script. Quoting some variables.

dvzrv committed:
tools/faust2appls/faust2jaqt: Quoting all variables.

dvzrv committed:
tools/faust2appls/faust2lv2: Quoting all variables. Setting shebang properly. Removing remaining non-qt5 parts.

sletz commented:
Thanks.

- **faust2jaqt** does not work anymore here on OS X. Does it works for you?
- why is "Quoting all variables" needed ?

dvzrv committed:
tools/faust2appls/faust2jaqt: Fixing shebang to use /usr/bin/env. Setting -e specifically.

sletz commented:
faust2jaqt  good.dsp
ERROR : cannot open file '' : No such file or directory

dvzrv commented:
Well, a lot of variables are not quoted and yet they might contain e.g. paths with whitespaces in them.
This is very dangerous, when e.g. removing files recursively, as this can lead to a lot of files being removed, that shouldn't be removed.
The easiest way to circumvent this is to quote variables. This also has the upside of being lintable using shellcheck :-)

dvzrv commented:
Do you have a good example to build using faust2jaqt? I have no macOS system available, so I can't really test that.
It's fairly possible, that I broke something.

dvzrv commented on tools/faust2appls/faust2jaqt:
```diff
@@ -151,38 +153,38 @@ for p in $FILES; do
-	    $QMAKE -project "QT += widgets printsupport network" "CONFIG+=warn_off" "$CLANGOPT" "INCLUDEPATH+=$CUR" "INCLUDEPATH+=$FAUSTINC /opt/local/include" "QMAKE_CXXFLAGS=$CXXFLAGS -Wno-unused-parameter $FAUSTTOOLSFLAGS" "LIBS+=$ARCHLIB $SOUNDFILELIBS $SAMPLERATELIBS $OSCLIBS $HTTPLIBS" "HEADERS+=$FAUSTINC/faust/gui/QTUI.h" "RESOURCES+= $FAUSTINC/faust/gui/Styles/Grey.qrc" "$OSCDEFS" "$HTTPDEFS" "$ ...
-	    $QMAKE $SPEC QMAKE_CFLAGS_ISYSTEM=-I
+	    "$QMAKE" -project "QT += widgets printsupport network" "CONFIG+=warn_off" "$CLANGOPT" "INCLUDEPATH+=$CUR" "INCLUDEPATH+=$FAUSTINC /opt/local/include" "QMAKE_CXXFLAGS=$CXXFLAGS -Wno-unused-parameter $FAUSTTOOLSFLAGS" "LIBS+=$ARCHLIB $SOUNDFILELIBS $SAMPLERATELIBS $OSCLIBS $HTTPLIBS" "HEADERS+=$FAUSTINC/faust/gui/QTUI.h" "RESOURCES+= $FAUSTINC/faust/gui/Styles/Grey.qrc" "$OSCDEFS" "$HTTPDEFS"  ...
+	    "$QMAKE" "$SPEC" QMAKE_CFLAGS_ISYSTEM=-I
```
It seems, that "$SPEC" is empty on all OS, but Darwin, which means, that it can not be quoted here easily.
Relying on variables in such a way is not robust and should be changed.

dvzrv commented on tools/faust2appls/faust2jaqt:
```diff
@@ -151,38 +153,38 @@ for p in $FILES; do
         fi
     else
-        faust -i -json -a $ARCHFILE $OPTIONS "$SRCDIR/$f" -o "$TMP/${f%.dsp}_tmp.cpp" || exit
+        faust -i -json -a "$ARCHFILE" "$OPTIONS" "$SRCDIR/$f" -o "$TMP/${f%.dsp}_tmp.cpp" || exit
```
Same as with `$SPEC`, `$OPTIONS` might be empty and therefore can not be quoted easily.
This is another case of unrobust variable use and should be changed.

dvzrv committed:
tools/faust2appls/faust2jaqt: Making sure, that OPTIONS are only used when converting from dsp to cpp, if they are actually defined.
Making sure, that SPEC is only used with qmake/qmake-qt5, if it is non-empty.

dvzrv commented:
@sletz Please try again with `faust2ajqt`. this works for me now!

dvzrv committed:
tools/faust2appls/faust2lv2: Providing CXXFLAGS and CPPFLAGS in arrays to be able to quote them properly.
Only optionally using OPTIONS or FAUSTTOOLSFLAGS when compiling, if they are non-empty.

dvzrv commented:
There were similar issues with how `CXXFLAGS` and `CPPFLAGS` were set in `faust2lv2`.

josmithiii commented:
I am getting normal results with faust2caqt and faust2jaqt on Mac OS X
running Qt 5.12.4_1 installed by MacPorts. There is an error message,
however:

Project WARNING: Qt has only been tested with version 10.14 of the
platform SDK, you're using 10.15.
Project WARNING: This is an unsupported configuration. You may
experience build issues, and by using
Project WARNING: the 10.15 SDK you are opting in to new features that
Qt has not been prepared for.
Project WARNING: Please downgrade the SDK you use to build your app to
version 10.14, or configure
Project WARNING: with CONFIG+=sdk_no_version_check when running qmake
to silence this warning.

On Sat, Sep 28, 2019 at 10:49 AM David Runge <notifications@github.com> wrote:
>
> There were similar issues with how CXXFLAGS and CPPFLAGS were set in faust2lv2.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or mute the thread.



-- 

Julius O. Smith III <jos@ccrma.stanford.edu>
Professor of Music and, by courtesy, Electrical Engineering
CCRMA, Stanford University
http://ccrma.stanford.edu/~jos/

dvzrv committed:
tools/faust2appls/faust2jaqt: Turning CXXFLAGS, FILES and OPTIONS into proper arrays, to which additions can just be added with the += operator. Using the arrays as separate elements with the [@] operator in e.g. faust compiler calls. Expanding to single string with [*] operator in qmake call.

dvzrv commented:
@josmithiii thanks for the feedback.
The warning message (should) be unrelated to my pull request though.

@sletz do you have an idea what `FAUSTTOOLSFLAGS` is used for and where it is defined? It's being used in the scripts, but it's not defined anywhere.

dvzrv committed:
tools/faust2appls/faust2lv2: Turning OPTIONS and FILES into arrays. Removing quoted versions of CPPFLAGS. Adding to CPPFLAGS with += operator. Using CPPFLAGS, and OPTIONS by expanding them as several strings with the [@] operator (e.g. in faust or CXX calls). Expanding CPPFLAGS with [*] to single string in calls to QMAKE (because they are being assigned to a variable).

dvzrv committed:
tools/faust2appls/faust2faustvst: Turning PROCARCH, OPTIONS, FILES, CXXFLAGS and CPPFLAGS into arrays, so they can be more easily handled and safely added to. Only using OPTIONS, if they are actually defined. Only using FAUSTTOOLSFLAGS, if they are actually defined. Dropping qt5 related extra check for x11extras and adding it directly to qmake call.

josmithiii commented:
It is not defined in my shell, so I think it is empty when used by the
script.
It looks to me like a hook for finding faust tools not installed in
/usr/local/bin, etc., if any.

On Sat, Sep 28, 2019 at 1:51 PM David Runge <notifications@github.com>
wrote:

> @josmithiii <https://github.com/josmithiii> thanks for the feedback.
> The warning message (should) be unrelated to my pull request though.
>
> @sletz <https://github.com/sletz> do you have an idea what FAUSTTOOLSFLAGS
> is used for and where it is defined? It's being used in the scripts, but
> it's not defined anywhere.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/grame-cncm/faust/pull/349?email_source=notifications&email_token=AAQZKFOJMIVAQVN7UAIVBFDQL67S5A5CNFSM4I3NS3T2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD73CFGI#issuecomment-536224409>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAQZKFK5VJHNACHZ5RVE5NTQL67S5ANCNFSM4I3NS3TQ>
> .
>


-- 

Julius O. Smith III <jos@ccrma.stanford.edu>
Professor of Music and, by courtesy, Electrical Engineering
CCRMA, Stanford University
http://ccrma.stanford.edu/~jos/ <http://ccrma.stanford.edu/>

dvzrv commented:
@josmithiii judging from the locations, it's supposed to carry flags. I just don't know why there's no mention of these anywhere else.
Probably a legacy artifact :)

dvzrv committed:
tools/faust2appls/faust2faustvst: Setting -eou bash options for more strict handling of undefined variables and return code handling. Guarding all potentially undefined variables.

dvzrv committed:
tools/faust2appls/faust2jaqt: Setting -eou pipefaul bash options. Guarding all potentially undefined variables and defining OSCLIBS (formerly wrongly defined as OSCLIB), OSCDEFS, HTTPLIBS, HTTPDEFS, QRDEFS, POLYDEFS and MIDIDEFS.
These definitions should go into a separate LIBS/DEFS array and plainly be added to on demand.

dvzrv committed:
tools/faust2appls/faust2lv2: Adding -eou pipefail bash options to have stricter handling of return codes and potentially undefined variables. Guarding all potentially undefined variables. Removing debug echo for dspname.

dvzrv commented:
@sletz I've rebased against current master-dev. Please check and feel free to merge at any time.
